src/bots/liquidator.ts

```diff
@@ -105,114 +320,30 @@ export class LiquidatorBot implements Bot {
 						if (liquidateePosition.baseAssetAmount.isZero()) {
 							continue;
 						}
-						const liquidatorPosition = liquidatorUser.getUserPosition(
-							liquidateePosition.marketIndex
-						);
-
-						let currentPosBaseAmount = new BN(0);
-						if (liquidatorPosition !== undefined) {
-							currentPosBaseAmount = liquidatorPosition.baseAssetAmount;
-						}
-
-						const market = this.clearingHouse.getMarketAccount(
-							liquidateePosition.marketIndex
-						);
 
-						logger.info(
-							`  liquidating position in market ${liquidateePosition.marketIndex.toString()}, size: ${convertToNumber(
-								liquidateePosition.baseAssetAmount,
-								BASE_PRECISION
-							).toString()}`
-						);
-						logger.info(
-							`    liquidatorPosition0: ${convertToNumber(
-								currentPosBaseAmount,
-								BASE_PRECISION
-							).toString()}`
-						);
-						const newPosition: UserPosition = {
-							baseAssetAmount: currentPosBaseAmount.add(
-								liquidateePosition.baseAssetAmount
-							),
-							lastCumulativeFundingRate:
-								liquidatorPosition?.lastCumulativeFundingRate,
-							marketIndex: liquidatorPosition?.marketIndex,
-							quoteAssetAmount: liquidatorPosition?.quoteAssetAmount,
-							quoteEntryAmount: liquidatorPosition?.quoteEntryAmount,
-							openOrders: liquidatorPosition
-								? liquidatorPosition.openOrders
-								: new BN(0),
-							openBids: liquidatorPosition
-								? liquidatorPosition.openBids
-								: new BN(0),
-							openAsks: liquidatorPosition
-								? liquidatorPosition.openAsks
-								: new BN(0),
-							realizedPnl: new BN(0),
-							lpShares: new BN(0),
-							lastFeePerLp: new BN(0),
-							lastNetBaseAssetAmountPerLp: new BN(0),
-							lastNetQuoteAssetAmountPerLp: new BN(0),
-						};
-						logger.info(
-							`    liquidatorPosition1: ${convertToNumber(
-								newPosition.baseAssetAmount,
-								BASE_PRECISION
-							).toString()}`
+						const baseAmountToLiquidate = this.calculateBaseAmountToLiquidate(
+							liquidatorUser,
+							liquidateePosition
 						);
+						const sufficientMarginToLiquidate =
```

Comment:
if you already calculated baseAmountToLiquidate based on the free collateral, do you really need to do `baseAmountToLiquidate`?